src/libCli/MessageParsing.cpp

```diff
@@ -36,202 +37,211 @@ int parseFieldValue(ParsedElement & f_parseTree, google::protobuf::Message * f_m
-                {
-                if(f_isRepeated)
-                {
-                    reflection->AddAllocatedMessage(f_message, f_fieldDescriptor, subMessage.release());
+                    if(f_isRepeated)
+                    {
+                        reflection->AddString(f_message, f_fieldDescriptor, resultString);
+                    }
+                    else
+                    {
+                        reflection->SetString(f_message, f_fieldDescriptor, resultString);
+                    }
                 }
                 else
                 {
-                    reflection->SetAllocatedMessage(f_message, subMessage.release(), f_fieldDescriptor);
-                }
+                    // otherwise we directly parse the given string:
+                    if(f_isRepeated)
+                    {
+                        reflection->AddString(f_message, f_fieldDescriptor, valueString);
+                    }
+                    else
+                    {
+                        reflection->SetString(f_message, f_fieldDescriptor, valueString);
+                    }
                 }
-                else
+                break;
+            case google::protobuf::FieldDescriptor::CppType::CPPTYPE_MESSAGE:
                 {
-                    std::cerr << "Error parsing sub-message for field '" << f_fieldDescriptor->name() << "'" << std::endl;
-                    return -1;
+                    const google::protobuf::Descriptor * subMessageDescriptor = f_fieldDescriptor->message_type();
+                    std::unique_ptr<google::protobuf::Message> subMessage = parseMessage(f_parseTree, f_factory, subMessageDescriptor);
+                    if(subMessage != nullptr)
+                    {
+                        if(f_isRepeated)
+                        {
+                            reflection->AddAllocatedMessage(f_message, f_fieldDescriptor, subMessage.release());
+                        }
+                        else
+                        {
+                            reflection->SetAllocatedMessage(f_message, subMessage.release(), f_fieldDescriptor);
+                        }
+                    }
+                    else
+                    {
+                        std::cerr << "Error parsing sub-message for field '" << f_fieldDescriptor->name() << "'" << std::endl;
+                        return -1;
+                    }
                 }
-            }
-            break;
-        default:
-            std::cerr << "Error: Parsing Field '" << f_fieldDescriptor->name() << "'. It has the unsupported type: '" << f_fieldDescriptor->type_name() << "'" << std::endl;
-            return -1;
+                break;
+            default:
+                std::cerr << "Error: Parsing Field '" << f_fieldDescriptor->name() << "'. It has the unsupported type: '" << f_fieldDescriptor->type_name() << "'" << std::endl;
+                return -1;
+        }
 
     }
+    catch(std::exception& e)
```

Comment:
only catch exception for integer conversions.
this also catches for enum, string, message types, which actually do not have any integer conversion.